### PR TITLE
Docs for pyarrow reader / writer

### PR DESCRIPTION
**Pull request by TomAugspurger**
This adds docs for the pyarrow reader / writer.

I think most users of this will benefit from it being the primary method of *producing* stac-geoparquet (because it has the best support for nested data).

For analytics, where all the features of a library like geopandas will be desired, I'll want to work on https://github.com/pandas-dev/pandas/issues/57411 which prevents (geo)pandas from reading these parquet files.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,40 +8,28 @@ This library helps convert [STAC Items](https://github.com/radiantearth/stac-spe
...
-
-```python
->>> import requests
->>> import stac_geoparquet
->>> item = requests.get("https://planetarycomputer.microsoft.com/api/stac/v1/collections/naip/items/ia_m_4209150_sw_15_060_20190828_20191105").json()
->>> df = stac_geoparquet.to_geodataframe([item])
->>> df.to_parquet("naip.parquet")
-```
+Use `stac_geoparquet.to_arrow.stac_items_to_arrow` and
+`stac_geoparquet.from_arrow.stac_table_to_items` to convert between STAC items
```
Comment (kylebarron) [inline README.md:12]: I didn't give a lot of thought to the public API here. Maybe the public API should condense to `stac_geoparquet.arrow`? The function names are quite verbose here.

File: README.md [hunk 2]
```diff
@@ -8,40 +8,28 @@ This library helps convert [STAC Items](https://github.com/radiantearth/stac-spe
...
+and Arrow tables. Arrow Tables of STAC items can be written to parquet with
+`stac_geoparquet.to_parquet.to_parquet`.
```
Comment (kylebarron) [inline README.md:14]: Likewise this is verbose, but for now this `to_parquet` only works with an Arrow table, so maybe this should live in an `arrow` module in `stac_geoparquet.arrow` as well?

**Comment by kylebarron** [comment]
> For analytics, where all the features of a library like geopandas will be desired, I'll want to work on [pandas-dev/pandas#57411](https://github.com/pandas-dev/pandas/issues/57411) which prevents (geo)pandas from reading these parquet files.

Interesting... I suppose I've never hit that because I'm not often saving data with the embedded Pandas-specific metadata. If you write this table without the Pandas-specific metadata it loads fine:

```py
import pyarrow as pa
import pandas as pd
import pyarrow.parquet as pq

list_int = pa.list_(pa.int64())
col = pa.array([[1, 1], [2, 2]])
table = pa.table({'col': col})
pq.write_table(table, "ex.parquet")

df = pq.read_table("ex.parquet").to_pandas(types_mapper=pd.ArrowDtype)
print(repr(df))
print(df.dtypes)
```

gives:
```
     col
0  [1 1]
1  [2 2]
```

```
col    list<element: int64>[pyarrow]
dtype: object
```

**Comment by TomAugspurger** [comment]
Oh, sorry, I must have been testing against the `dtype_backend="pyarrow"` implementation. That should produce the same output, aside from the pandas-specific metadata, which as you say is the issue.

```python
>>> geopandas.read_parquet("items.parquet").head()
```

works fine, so I think we can ignore that issue.

**Comment by kylebarron** [comment]
> must have been testing against the `dtype_backend="pyarrow"` implementation

I'm not 100% sure here, but I thought that I'm getting the same result by passing `types_mapper=pd.ArrowDtype` into the `pyarrow.Table.to_pandas` call. The output dtype is still pyarrow-backed:
```
col    list<element: int64>[pyarrow]
```

File: README.md (hunk 1 above)
Comment (kylebarron) [reply README.md:12]: Created https://github.com/stac-utils/stac-geoparquet/pull/47

Comment (TomAugspurger) [reply README.md:12]: Updated after #47.

**Comment by kylebarron** [comment]
Thanks!

**Pull request merged**